APWorld/ER_Rules.py

```diff
@@ -428,8 +436,10 @@ def __init__(self, world: MinitWorld) -> None:
             "boatguy watered":  self.helpers["wateringcan"],
             "left machine": lambda state:
                 RuleUtils.has_darkroom(self.player, state, 1, self.darkrooms)
-                and state.has("ItemCoffee", self.player),
-            "right machine":  self.helpers["sword"],
+                and state.has_all({"ItemSwim", "ItemCoffee"}, self.player),
+            "right machine": lambda state:
+                RuleUtils.has_darkroom(self.player, state, 1, self.darkrooms)
+                and state.has("has_sword", self.player),
```

Comment:
```suggestion
                self.helpers["darkroom1"] and state.has("has_sword", self.player),
```
also noticed there's a previously existing has_darkroom without using self.helpers feel free to refactor that or leave it as-is

Reply:
Needs (state) as a parameter added, but this got worked in in a push that came later.